packages/utils/interactive-hooks/src/useAsPressable.ts

```diff
@@ -146,3 +150,21 @@ export function useHoverState<T extends object>(props: IWithPressableOptions<T>)
   const [hoverProps, hoverState] = useHoverHelper(props);
   return [{ ...props, ...usePressability({ ...props, ...hoverProps }) }, hoverState];
 }
+
+/**
+ * This routine hooks the props to pass to a Pressable component to obtain the current state of the Pressable as well as generating
+ * state change updates when those props change. This allows a parent component to control the render of the whole component rather than having
+ * to split the code between a child function or style function.
+ *
+ * @param props - props to pass to a Pressable component
+ * @returns - modified props to pass into the Pressable as well as the current state with regards to hover, focus, and press
+ */
+export function usePressableState(props: PressablePropsExtended): { props: PressableProps; state: IPressableState } {
+  const { onPressIn, onPressOut, onHoverIn, onHoverOut, onFocus, onBlur, ...rest } = props;
+  const [focusProps, focusState] = useFocusHelper({ onFocus, onBlur });
+  const [pressProps, pressState] = usePressHelper({ onPressIn, onPressOut });
+  const [hoverProps, hoverState] =
+    Platform.OS !== 'android' && Platform.OS !== 'ios' ? useHoverHelper({ onHoverIn, onHoverOut }) : [{}, {}];
```

Comment:
nit: Maybe something like this?

```suggestion
const platformSupportsHover = (Platform.OS !== 'android' && Platform.OS !== 'ios');
const const [hoverProps, hoverState] = platformSupportsHover ? useHoverHelper({ onHoverIn, onHoverOut }) : [{}, {}];
```